appointment-svc/middlewares/auth-middleware.ts

```diff
@@ -1,21 +1,51 @@
 import jwksRsa from 'jwks-rsa';
 import dotenv from 'dotenv';
+import { NextFunction, Request, Response } from 'express';
+import decryptData from '../utils/decryption';
 const { expressjwt: jwt } = require('express-jwt');
 
 dotenv.config();
 
-const authMiddleware = () => {
-  return jwt({
+export interface IMiddlewareRequest extends Request {
+  decryptedData: any;
```

Comment:
the object that I encrypt has a fixed structure, so I think we could type it later

Reply:
And also I think this type will be often reused (in all the endpoints which use this middleware to type the req), so it would need to be in the types file